src/resolvers/helpers/filterOperators.ts

```diff
@@ -1,30 +1,80 @@
 /* eslint-disable no-use-before-define */
 
-import { getNamedType, GraphQLInputType } from 'graphql-compose/lib/graphql';
-import type { Model } from 'mongoose';
-import type {
-  InputTypeComposer,
-  InputTypeComposerFieldConfigAsObjectDefinition,
-} from 'graphql-compose';
-import { upperFirst, getIndexedFieldNamesForGraphQL } from '../../utils';
+import {
+  getNamedType,
+  GraphQLInputType,
+  GraphQLInputObjectType,
+  GraphQLScalarType,
+  GraphQLEnumType,
+} from 'graphql-compose/lib/graphql';
+import type { Model, Schema, SchemaType, VirtualType } from 'mongoose';
+import type { InputTypeComposer } from 'graphql-compose';
+import type { InputTypeComposerFieldConfigAsObjectDefinition } from 'graphql-compose';
+
+import { upperFirst } from '../../utils';
 import type { FilterHelperArgsOpts } from './filter';
 
-export type FilterOperatorNames = 'gt' | 'gte' | 'lt' | 'lte' | 'ne' | 'in[]' | 'nin[]';
-const availableOperators: FilterOperatorNames[] = ['gt', 'gte', 'lt', 'lte', 'ne', 'in[]', 'nin[]'];
+export type FilterOperatorNames =
+  | 'gt'
+  | 'gte'
+  | 'lt'
+  | 'lte'
+  | 'ne'
+  | 'in[]'
+  | 'nin[]'
+  | 'regex'
+  | 'exists';
+
+export enum availableOperators {
+  GT = 'gt',
+  GTE = 'gte',
+  LT = 'lt',
+  LTE = 'lte',
+  NE = 'ne',
+  IN = 'in',
+  NIN = 'nin',
+  REGEX = 'regex',
+  EXISTS = 'exists',
+}
+
+export type FilterOperatorsOpts =
+  | {
+      [fieldName: string]: FilterOperatorsOpts | FilterOperatorNames[] | false;
+    }
+  | false;
 
 export const OPERATORS_FIELDNAME = '_operators';
 
-export type FilterOperatorsOpts = {
-  [fieldName: string]: FilterOperatorNames[] | false;
-};
+export function _createRegexInput(itc: InputTypeComposer<any>): void {
+  itc.schemaComposer.getOrCreateITC(`${OPERATORS_FIELDNAME}RegexInput`, (tc) => {
+    tc.addFields({
+      match: {
+        name: `${OPERATORS_FIELDNAME}RegexStringInput`,
+        type: 'String!',
+      },
+      options: {
+        name: `${OPERATORS_FIELDNAME}RegexOptionsInput`,
+        type: 'String',
+      },
+    });
```

Comment:
What do you think if we create custom scalar `RegExp` which will be string representation of RegExp?

```js
const r = new RegExp('ABC.*', 'g');
console.log(r.toString()); 
// outputs "/ABC.*/g"
```

So we can provide to user the following API:
```graphql
query {
  user(nameMask: "/ABC.*/i") {
    name
    age
  }
}
```

instead of current implementation 
```graphql
query {
  user(nameMask: { match: "ABC.*",  options: "i" }) {
    name
    age
  }
}
```

Reply:
Simple `RegExp` custom scalar type implementation:

```ts
import { GraphQLScalarType, Kind } from 'graphql-compose/lib/graphql';

function parseStringWithRegExp(str: string): RegExp {
  if (str.startsWith('/')) {
    const m = str.match(/^\/(.+)\/([gimsuy]*)$/);
    if (m) {
      return new RegExp(m[1], m[2]);
    }
    throw new TypeError('Field error: cannot parse provided string as RegExp object');
  } else {
    // simple regexp without expression flags
    return new RegExp(str);
  }
}

const GraphQLRegExp = new GraphQLScalarType({
  name: 'RegExp',
  // TODO: add link to RegExp spec
  description: 'The string representation of JavaScript regexp.',
  serialize: String,
  parseValue(value: any) {
    if (typeof value !== 'string') {
      throw new TypeError('Field error: GraphQL RegExp value should be provided as a string');
    }
    return parseStringWithRegExp(value);
  },
  parseLiteral(ast) {
    if (ast.kind !== Kind.STRING) {
      throw new TypeError('Field error: GraphQL RegExp value should be provided as a string');
    }
    return parseStringWithRegExp(ast.value);
  },
});

export default GraphQLRegExp;
```

Reply:
Yeah, let's run with that.